homeassistant/components/peco/sensor.py

```diff
@@ -46,11 +46,11 @@ async def async_setup_entry(
     websession = hass.data[DOMAIN][config_entry.entry_id]["websession"]
     conf: MappingProxyType[str, Any] = config_entry.data
 
-    async def async_update_data():
+    async def async_update_data() -> dict:
         """Fetch data from API."""
         if conf["county"] == "TOTAL":
             try:
-                data = await api.get_outage_totals(websession)
+                data: dict = await api.get_outage_totals(websession)
```

Comment:
To typecheck this your pypi package would have to be type annotated too, and it would have to export its type definitions. 
https://mypy.readthedocs.io/en/stable/running_mypy.html?highlight=py.typed#how-mypy-handles-imports

You may use `cast(dict[str, int], await api.get_outage_totals(websession))` (`from typing import cast`) or `# type: ignore[no-any-return]` to skip the check.

Same for line 64